src/drivers/hardware.h

```diff
@@ -2,6 +2,19 @@
 
 #include "types.h"
 
-extern u16 monitor, keyboard, clock;
+#define LEM 0x7349F615
+#define KEYBOARD 0x30CF7406
+#define CLOCK 0x12D0B402
 
-void hardwareLoop();
+struct HardwareInfo;
+
+struct HardwareList {
+    u16 size;
+    struct HardwareInfo *hardware;
+};
+
+struct HardwareList hardwareList = {0, 0};
```

Comment:
just move this line in the .c ^^
with maybe an extern struct... if it's used in an other file.
I will add .lcomm to my assembler